### DISCUSSION TITLE
Smart AI Camera Search

### DISCUSSION THREAD

**Discussion by faridalizade** [inspireui/support#5680] [New Feature/Ideas Requests]
Hello,

I have a question, I want to add AI search for all products on our platform. E.g. Anyone can capture or draw a product image to search it. We do not use Barcode scanner or similar things.

On the web version I see Impresee (impresee.com) that doing the same as I ask.

Thank you.

**Comment by ecomdeveloperbd** [comment]
This feature is also very important for present times

**Comment by krrisharora** [comment]
Yes, this is already coming in all ecommerce apps in India today. Examples like: Meesho, Glowroad, Amazon, Flipkart, Shopsy and many more. No one uses Barcode scanner in India to search products. We need Image search

**Comment by ecomdeveloperbd** [comment]
Nowadays, Image Searching is Trending feature.

**Reply by usamaahmed5656** [reply]
Many of guys is asking for a AI camera Search these days on your discussion platform on github..

the best and easy way to integrate this is 


1)use google cloud vision api

2) User click a picture from app which will be send to google vision via its api ..

3)google vision will analyze that image and generate labels matching to that image

4)app receives thoes labels and send it to woocommerce search 

5)finally woocommerce search will generate search result based on the labels

6) and we can easily show such results in our app generated by woo

**Comment by usamaahmed5656** [comment]
Many of guys is asking for a AI camera Search these days on your discussion platform on github..

the best and easy way to integrate this is 


1)use google cloud vision api

2) User click a picture from app which will be send to google vision via its api ..

3)google vision will analyze that image and generate labels matching to that image

4)app receives thoes labels and send it to woocommerce search 

5)finally woocommerce search will generate search result based on the labels

6) and we can easily show such results in our app generated by woo
